tests/ui/use_self.rs

```diff
@@ -79,6 +79,7 @@ mod issue2894 {
     }
 
     // This should not be linted
```

Comment:
Same here. I get that it's probably a different lint that should not trigger, but it still feels wrong.

Reply:
Yes, the comment says that `use_self` shouldn't trigger. 
I added the `#[allow(clippy::wrong_self_convention)]` to suppress the warning that occurred there, after current changes. I wasn't sure whether I should adjust the self convention there.
Agree it's better to adjust that uitest to the proper self convention, which I did in the next commit.